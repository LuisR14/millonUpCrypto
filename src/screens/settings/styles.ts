import {StyleSheet} from 'react-native';
import {dimension} from '../../themes/dimension';
import fontSize from '../../themes/fontSize';
import colors from '../../themes/colors';
const {horizontalScale, verticalScale} = dimension();

export default StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    marginHorizontal: 20,
  },
  textTitle: {
    fontWeight: 'bold',
    paddingHorizontal: 10,
    marginTop: 20,
    fontSize: fontSize.xlarge,
  },
  textSubTitle: {
    fontWeight: 'bold',
    paddingHorizontal: 10,
    fontSize: fontSize.large,
    color: colors.black,
  },
  textSwitch: {
    paddingHorizontal: 10,
    fontSize: fontSize.normal,
    color: colors.black,
  },
  switchContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },

  settingComponent: {
    width: horizontalScale(330),
    height: verticalScale(60),
    borderRadius: 10,
    marginVertical: 8,
    padding: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    shadowOffset: {
      width: 0,
      height: 10,
    },
    shadowOpacity: 0.3,
    shadowRadius: 5,
    elevation: 10,
  },
});
