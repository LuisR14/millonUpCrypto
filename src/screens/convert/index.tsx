/* eslint-disable react-hooks/exhaustive-deps */
/* eslint-disable react/no-unstable-nested-components */
import {useRoute} from '@react-navigation/native';
import React, {useEffect, useState} from 'react';
import {Image, SafeAreaView, TouchableOpacity, View, Text} from 'react-native';
import styles from './styles';
import colors from '../../themes/colors';
import Margin from '../../components/margin';
import {useSelector} from 'react-redux';
import {CryptoProps} from '../../types/crypto';
import ChangeIcon from '../../assets/icons/change.svg';
import {CryptoSelectPicker} from '../../components/cryptoSelectPicker';

export const ConvertScreen = () => {
  const params = useRoute<any>();
  const cryptoValueParam = params?.params?.cryptoValue;
  const [cryptoFromValue, setcryptoFromValue] = useState('');
  const [items, setItems] = useState<any[]>([]);
  const [cryptoToValue, setcryptoToValue] = useState('');
  const [valueTo, setValueTo] = useState('1');
  const [valueFrom, setValueFrom] = useState(cryptoValueParam);
  const [openFrom, setOpenFrom] = useState(false);
  const [openTo, setOpenTo] = useState(false);
  const zIndex = 1;
  const borderBottomRadius = 0;
  const borderBottomRadiusClose = 8;
  const cryptoList = useSelector((state: any) => state.cryptoList);
  const Settings = useSelector((state: any) => state.userSettings);
  const Theme = Settings?.darkMode;

  useEffect(() => {
    if (cryptoList?.listBackup?.length > 0 && !cryptoValueParam) {
      setValueFrom(cryptoList?.listBackup[0].price_usd);
    }
  }, [cryptoList?.listBackup]);

  useEffect(() => {
    const cryptoListDropDown: React.SetStateAction<any[]> = [
      {
        id: 0,
        label: 'USD',
        value: '1',
        icon: () => (
          <Image
            source={{
              uri: 'https://coinicons-api.vercel.app/api/icon/usd',
            }}
            style={styles.cryptoIcon}
          />
        ),
      },
    ];
    cryptoList?.listBackup?.map((cryptoItem: CryptoProps) => {
      cryptoListDropDown.push({
        id: cryptoItem?.id,
        label: cryptoItem?.symbol,
        value: cryptoItem?.price_usd,
        icon: () => (
          <Image
            source={{
              uri: `https://coinicons-api.vercel.app/api/icon/${cryptoItem?.symbol?.toLowerCase()}`,
            }}
            style={styles.cryptoIcon}
          />
        ),
      });
    });
    setItems(cryptoListDropDown);
  }, []);

  useEffect(() => {
    const cryptoValueNumber = parseFloat(cryptoFromValue);
    const valueFromNumber = parseFloat(valueFrom);
    const valueToNumber = parseFloat(valueTo);
    if (cryptoFromValue) {
      if (valueToNumber === 1) {
        setcryptoToValue((cryptoValueNumber * valueFromNumber).toFixed(5));
      } else {
        setcryptoToValue(
          ((valueFromNumber / valueToNumber) * cryptoValueNumber).toFixed(5),
        );
      }
    } else {
      setcryptoToValue('0.00');
    }
  }, [cryptoFromValue, valueFrom, valueTo]);

  const onChangeOrder = () => {
    setValueFrom(valueTo);
    setValueTo(valueFrom);
  };

  return (
    <SafeAreaView testID="safeAreaViewTest" style={styles.container}>
      <Text
        testID="titleConvertScreenTest"
        style={{
          ...styles.textTitle,
          color: Theme !== 'dark' ? colors.black : colors.white,
        }}>
        Convert
      </Text>
      <View
        testID="viewTextInputTest"
        style={{
          ...styles.textInputContainer,
          zIndex,
          borderBottomLeftRadius: openFrom
            ? borderBottomRadius
            : borderBottomRadiusClose,
          backgroundColor: Theme !== 'dark' ? colors.lightGray : colors.gray,
        }}>
        <CryptoSelectPicker
          testID="selectPickerUp"
          editable={true}
          openFrom={openFrom}
          valueFrom={valueFrom}
          items={items}
          setOpenFrom={setOpenFrom}
          setValueFrom={setValueFrom}
          setItems={setItems}
          cryptoFromValue={cryptoFromValue}
          setcryptoFromValue={setcryptoFromValue}
        />
      </View>
      <Margin bottom={40} />
      <TouchableOpacity onPress={onChangeOrder}>
        <ChangeIcon fill={Theme !== 'dark' ? colors.black : colors.white} />
      </TouchableOpacity>
      <Margin bottom={20} />
      <View
        testID="viewTextInput2Test"
        style={{
          ...styles.textInputContainer,
          borderBottomLeftRadius: openTo
            ? borderBottomRadius
            : borderBottomRadiusClose,
          backgroundColor: Theme !== 'dark' ? colors.lightGray : colors.gray,
        }}>
        <CryptoSelectPicker
          testID="selectPickerDown"
          editable={false}
          openFrom={openTo}
          valueFrom={valueTo}
          items={items}
          setOpenFrom={setOpenTo}
          setValueFrom={setValueTo}
          setItems={setItems}
          cryptoFromValue={cryptoToValue}
        />
      </View>
    </SafeAreaView>
  );
};
