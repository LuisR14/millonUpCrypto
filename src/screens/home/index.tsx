/* eslint-disable react-hooks/exhaustive-deps */
import React, {useCallback, useEffect, useRef, useState} from 'react';
import {
  Modal,
  RefreshControl,
  Text,
  TextInput,
  TouchableHighlight,
  TouchableOpacity,
  View,
} from 'react-native';
import {CryptoProps} from '../../types/crypto';
import styles from './styles';
import {RenderFooter} from '../../components/flatlistFooter';
import CryptoComponent from '../../components/cryptoList';
import Margin from '../../components/margin';
import SearchIcon from '../../assets/icons/search.svg';
import CleanSearchIcon from '../../assets/icons/cleanSearch.svg';
import colors from '../../themes/colors';
import {getCryptoList} from '../../functions/getCryptoList';
import {loadMoreCrypto} from '../../functions/loadMoreCrypto';
import {searchText} from '../../functions/searchText';
import {SkeletonLoader} from '../../components/skeletonLoader';
import {ModalDetailComponent} from '../../components/modalDetail';
import {useDispatch, useSelector} from 'react-redux';
import {
  create,
  createBackup,
  update,
  updateBackup,
} from '../../redux/cryptoListSlice';
import {CryptoListEmpty} from '../../components/cryptoListEmpty';
import {sleep} from '../../functions/sleep';
import Animated, {
  useAnimatedScrollHandler,
  useSharedValue,
} from 'react-native-reanimated';
import {SafeAreaView} from 'react-native';
import SplashScreen from 'react-native-splash-screen';
import {filter} from '../../functions/filterCryptoList';
import {saveCryptoList} from '../..//functions/saveCryptoList';
import {getCryptoListSaved} from '../..//functions/getCryptoListSaved';
import {useNetInfo} from '@react-native-community/netinfo';
import {useToast} from 'react-native-toast-notifications';

export const HomeScreen = () => {
  const [cryptoDetail, setcryptoDetail] = useState<CryptoProps>();
  const [isLoading, setisLoading] = useState(false);
  const [isRefreshing, setisRefreshing] = useState(false);
  const [isLoadingMore, setisLoadingMore] = useState(false);
  const [count, setcount] = useState(1);
  const [search, setsearch] = useState('');
  const [modalDetail, setmodalDetail] = useState(false);
  const [blockUpdate, setblockUpdate] = useState(true);
  const [activefilterGainer, setActivefilterGainer] = useState(false);
  const [activefilterLosers, setActivefilterLosers] = useState(false);
  const [activefilterVolume24, setActivefilterVolume24] = useState(false);
  const [errorMsg, setErrorMsg] = useState(null);
  const dispatch = useDispatch();
  const searchRef = useRef<any>();
  const flatListRef = useRef<any>();
  const cryptoList = useSelector((state: any) => state.cryptoList);
  const Settings = useSelector((state: any) => state.userSettings);
  const Theme = Settings?.darkMode;
  const netInfo = useNetInfo();
  const Offline = Settings?.offline;
  const scrollY = useSharedValue(0);
  const toast = useToast();
  useEffect(() => {
    SplashScreen?.hide();
    setisLoading(true);
  }, []);

  useEffect(() => {
    if (netInfo.isConnected !== null) {
      checkCryptoApi();
    }
  }, [netInfo.isConnected]);

  useEffect(() => {
    if (Offline) {
      setblockUpdate(true);
      getSavedCryptoList();
    } else if (!Offline && netInfo.isConnected !== null) {
      setblockUpdate(false);
      checkCryptoApi();
    }
  }, [Offline]);

  useEffect(() => {
    if (errorMsg) {
      toast.show(errorMsg, {
        type: 'danger',
      });
      setErrorMsg(null);
    }
  }, [toast, errorMsg]);

  const checkCryptoApi = async () => {
    cleanFilter();
    if (Offline || netInfo.isConnected === false) {
      getSavedCryptoList();
    } else {
      const cryptoListResponse = await getCryptoList({
        setsearch,
      });
      if (cryptoListResponse?.length > 0) {
        dispatch(create(cryptoListResponse));
        dispatch(createBackup(cryptoListResponse));
        saveCryptoList(cryptoListResponse);
        setblockUpdate(false);
      } else {
        setErrorMsg('server error');
      }
    }
    setisLoading(false);
  };
  const updateCryptoList = async () => {
    const cryptoListUpdate = await loadMoreCrypto({
      setisLoadingMore,
      count,
      setcount,
      search,
    });
    if (cryptoListUpdate.length > 0) {
      dispatch(update(cryptoListUpdate));
      dispatch(updateBackup(cryptoListUpdate));
    } else {
      setErrorMsg('server error');
    }
  };

  const getSavedCryptoList = async () => {
    const list = await getCryptoListSaved();
    dispatch(create(list));
    dispatch(createBackup(list));
  };

  const handler = useAnimatedScrollHandler({
    onScroll: event => {
      scrollY.value = event.contentOffset.y;
    },
  });

  const scollToTop = () => {
    flatListRef.current.scrollToOffset({
      animated: true,
      offset: 0,
    });
  };

  const cleanFilter = () => {
    setActivefilterGainer(false);
    setActivefilterLosers(false);
    setActivefilterVolume24(false);
  };

  const keyExtractor = useCallback((item: CryptoProps) => item.id, []);

  const renderItems = useCallback(
    ({item, index}: any) => (
      <TouchableHighlight
        testID="buttonDetailsTest"
        style={styles.touchable}
        activeOpacity={0.6}
        underlayColor="lightgray"
        onPress={() => {
          setcryptoDetail(item);
          setmodalDetail(true);
        }}>
        <CryptoComponent cryptoItem={item} scrollY={scrollY} index={index} />
      </TouchableHighlight>
    ),
    [],
  );

  return (
    <>
      <SafeAreaView style={styles.container}>
        {isLoading && !isRefreshing ? (
          <SkeletonLoader />
        ) : (
          <>
            <View testID="viewTextInputTest" style={styles.textInputContainer}>
              <TouchableOpacity
                style={styles.touchableIconSearch}
                onPress={() => searchRef.current.focus()}>
                <SearchIcon />
              </TouchableOpacity>
              <TextInput
                testID="textInputTest"
                ref={searchRef}
                style={styles.textInput}
                placeholder="Search coin"
                placeholderTextColor={colors.gray}
                value={search}
                onChangeText={text => {
                  scollToTop();
                  cleanFilter();
                  const cryptoListFilter = searchText({
                    text,
                    setsearch,
                    cryptoList,
                  });
                  if (cryptoListFilter) {
                    dispatch(create(cryptoListFilter));
                  } else {
                    dispatch(create(cryptoList?.listBackup));
                  }
                }}
              />
              {search.length > 0 && (
                <TouchableOpacity
                  style={styles.touchableIconSearch}
                  onPress={() => {
                    setsearch('');
                    dispatch(create(cryptoList?.listBackup));
                  }}>
                  <CleanSearchIcon />
                </TouchableOpacity>
              )}
            </View>
            <Margin bottom={15} />
            <View style={styles.filtersContainer}>
              <TouchableOpacity
                testID="buttonFilterGainers"
                onPress={() => {
                  scollToTop();
                  if (activefilterGainer) {
                    setblockUpdate(false);
                    dispatch(create(cryptoList?.listBackup));
                  } else {
                    setActivefilterLosers(false);
                    setActivefilterVolume24(false);
                    setblockUpdate(true);
                    const filterlist = filter(
                      cryptoList?.listBackup,
                      'gainers',
                    );
                    dispatch(create(filterlist));
                  }
                  setActivefilterGainer(!activefilterGainer);
                }}
                style={{
                  ...styles.filtersButton,
                  backgroundColor: activefilterGainer
                    ? Theme === 'dark'
                      ? colors.white
                      : colors.lightBlue
                    : Theme === 'dark'
                    ? colors.gray
                    : colors.lightGray,
                }}>
                <Text style={styles.textFilters}>Gainers</Text>
              </TouchableOpacity>
              <TouchableOpacity
                testID="buttonFilterLosers"
                onPress={() => {
                  scollToTop();
                  if (activefilterLosers) {
                    setblockUpdate(false);
                    dispatch(create(cryptoList?.listBackup));
                  } else {
                    setActivefilterGainer(false);
                    setActivefilterVolume24(false);
                    setblockUpdate(true);
                    const filterlist = filter(cryptoList?.listBackup, 'losers');
                    dispatch(create(filterlist));
                  }
                  setActivefilterLosers(!activefilterLosers);
                }}
                style={{
                  ...styles.filtersButton,
                  backgroundColor: activefilterLosers
                    ? Theme === 'dark'
                      ? colors.white
                      : colors.lightBlue
                    : Theme === 'dark'
                    ? colors.gray
                    : colors.lightGray,
                }}>
                <Text style={styles.textFilters}>Losers</Text>
              </TouchableOpacity>
              <TouchableOpacity
                testID="buttonFilterVolume24"
                onPress={() => {
                  scollToTop();
                  if (activefilterVolume24) {
                    setblockUpdate(false);
                    dispatch(create(cryptoList?.listBackup));
                  } else {
                    setActivefilterLosers(false);
                    setActivefilterGainer(false);
                    setblockUpdate(true);
                    const filterlist = filter(
                      cryptoList?.listBackup,
                      'gainers',
                    );
                    dispatch(create(filterlist));
                  }
                  setActivefilterVolume24(!activefilterVolume24);
                }}
                style={{
                  ...styles.filtersButton,
                  backgroundColor: activefilterVolume24
                    ? Theme === 'dark'
                      ? colors.white
                      : colors.lightBlue
                    : Theme === 'dark'
                    ? colors.gray
                    : colors.lightGray,
                }}>
                <Text style={styles.textFilters}>Volume24h</Text>
              </TouchableOpacity>
            </View>
            <Margin bottom={10} />
            <Animated.FlatList
              testID="flatListTest"
              ref={flatListRef}
              contentContainerStyle={styles.flatlistContainer}
              keyboardDismissMode="on-drag"
              ListEmptyComponent={CryptoListEmpty}
              onScroll={handler}
              maxToRenderPerBatch={40}
              onEndReached={async () => {
                if (!blockUpdate) {
                  setblockUpdate(true);
                  updateCryptoList();
                  await sleep(5000);
                  setblockUpdate(false);
                }
              }}
              onEndReachedThreshold={1}
              refreshControl={
                <RefreshControl
                  refreshing={isRefreshing}
                  onRefresh={() => {
                    setisRefreshing(true);
                    setTimeout(() => {
                      setisRefreshing(false);
                    }, 400);
                    checkCryptoApi();
                  }}
                  title="Refreshing"
                  tintColor={Theme === 'dark' ? colors.white : colors.black}
                  titleColor={Theme === 'dark' ? colors.white : colors.black}
                  colors={
                    Theme === 'dark' ? [colors.lightBlue] : [colors.black]
                  }
                />
              }
              data={cryptoList?.list}
              renderItem={renderItems}
              keyExtractor={keyExtractor}
              ListFooterComponent={() => RenderFooter(isLoadingMore)}
            />
            <Modal
              testID="modalTest"
              style={{
                backgroundColor:
                  Theme !== 'dark' ? colors.white : colors.darkCyan,
              }}
              animationType="slide"
              visible={modalDetail}>
              <ModalDetailComponent
                cryptoItem={cryptoDetail}
                setmodalDetail={setmodalDetail}
              />
            </Modal>
          </>
        )}
      </SafeAreaView>
    </>
  );
};
