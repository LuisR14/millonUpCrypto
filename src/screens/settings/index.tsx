import React, {useState} from 'react';
import {SafeAreaView, Switch, View} from 'react-native';
import {Text} from 'react-native';
import styles from './styles';
import Margin from '../../components/margin';
import {useDispatch, useSelector} from 'react-redux';
import {updateUserOffline, updateUserTheme} from '../../redux/userSettingSlice';
import {saveUserTheme} from '../../functions/saveUserTheme';
import colors from '../../themes/colors';
import {useNetInfo} from '@react-native-community/netinfo';

export const SettingsScreen = () => {
  const dispatch = useDispatch();
  const Settings = useSelector((state: any) => state.userSettings);
  const Theme = Settings?.darkMode;
  const Offline = Settings?.offline;
  const [switchValue, setswitchValue] = useState(
    Theme === 'dark' ? true : false,
  );
  const netInfo = useNetInfo();

  const onChangeSwitch = (value: boolean) => {
    if (value) {
      setswitchValue(true);
      dispatch(updateUserTheme('dark'));
      saveUserTheme('dark');
    } else {
      setswitchValue(false);
      dispatch(updateUserTheme('light'));
      saveUserTheme('light');
    }
  };
  const onChangeSwitchOffline = (value: boolean) => {
    if (value) {
      dispatch(updateUserOffline(value));
    } else {
      dispatch(updateUserOffline(value));
    }
  };

  return (
    <SafeAreaView testID="safeAreaViewTest" style={styles.container}>
      <Text
        testID="textTitleTest"
        style={{
          ...styles.textTitle,
          color: Theme !== 'dark' ? colors.black : colors.white,
        }}>
        Settings
      </Text>
      <Margin bottom={20} />
      <View
        style={{
          ...styles.settingComponent,
          backgroundColor: Theme !== 'dark' ? colors.lightBlue : colors.white,
        }}>
        <Text testID="darkModeTextSettingTest" style={styles.textSubTitle}>
          Dark mode
        </Text>
        <View testID="viewSwitch1Test" style={styles.switchContainer}>
          <Switch
            testID="switch1Test"
            onValueChange={value => onChangeSwitch(value)}
            value={switchValue}
          />
        </View>
      </View>
      <View
        style={{
          ...styles.settingComponent,
          backgroundColor: Theme !== 'dark' ? colors.lightBlue : colors.white,
        }}>
        <Text testID="offlineModeTextSettingTest" style={styles.textSubTitle}>
          Offline mode
        </Text>
        <View testID="viewSwitch2Test" style={styles.switchContainer}>
          <Switch
            testID="switch2Test"
            disabled={!netInfo.isConnected}
            onValueChange={value => onChangeSwitchOffline(value)}
            value={Offline}
          />
        </View>
      </View>
    </SafeAreaView>
  );
};
