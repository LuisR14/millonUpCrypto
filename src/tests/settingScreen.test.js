import React from 'react';
import {renderWithProviders} from './utils_for_tests';
import {NavigationContainer} from '@react-navigation/native';
import {SettingsScreen} from '../screens/settings';
import {screen} from '@testing-library/react-native';

describe('SettingScreen', () => {
  let component;
  beforeEach(() => {
    component = renderWithProviders(
      <NavigationContainer>
        <SettingsScreen />
      </NavigationContainer>,
    );
  });

  it('snapshot of screen to JSON', () => {
    const {toJSON} = renderWithProviders(
      <NavigationContainer>
        <SettingsScreen />
      </NavigationContainer>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly setting screen', () => {
    expect(component).toBeTruthy();
  });

  it('renders correctly the safe area view', () => {
    const item = screen.getByTestId('safeAreaViewTest');
    expect(item).toBeOnTheScreen();
  });

  it('renders correctly the title text setting', () => {
    const item = screen.getByTestId('textTitleTest');
    expect(item).toBeOnTheScreen();
  });

  it('renders correctly the text setting dark mode', () => {
    const item = screen.getByTestId('darkModeTextSettingTest');
    expect(item).toBeOnTheScreen();
  });

  it('renders correctly the first switch view', () => {
    const item = screen.getByTestId('viewSwitch1Test');
    expect(item).toBeOnTheScreen();
  });
  it('renders correctly the first switch component', () => {
    const item = screen.getByTestId('switch1Test');
    expect(item).toBeOnTheScreen();
  });

  it('renders correctly the text setting offline mode', () => {
    const item = screen.getByTestId('offlineModeTextSettingTest');
    expect(item).toBeOnTheScreen();
  });

  it('renders correctly the second switch view', () => {
    const item = screen.getByTestId('viewSwitch2Test');
    expect(item).toBeOnTheScreen();
  });

  it('renders correctly the second switch component', () => {
    const item = screen.getByTestId('switch2Test');
    expect(item).toBeOnTheScreen();
  });
});
