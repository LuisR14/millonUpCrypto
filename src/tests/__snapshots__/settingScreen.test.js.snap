// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingScreen snapshot of screen to JSON 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "marginHorizontal": 20,
    }
  }
  testID="safeAreaViewTest"
>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginTop": 20,
        "paddingHorizontal": 10,
      }
    }
    testID="textTitleTest"
  >
    Settings
  </Text>
  <View
    style={
      {
        "marginBottom": 20,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "lightblue",
        "borderRadius": 10,
        "elevation": 10,
        "flexDirection": "row",
        "height": 98.57142857142857,
        "justifyContent": "space-between",
        "marginVertical": 8,
        "padding": 10,
        "shadowOffset": {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 5,
        "width": 660,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
          "paddingHorizontal": 10,
        }
      }
      testID="darkModeTextSettingTest"
    >
      Dark mode
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
      testID="viewSwitch1Test"
    >
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "height": 31,
            "width": 51,
          }
        }
        testID="switch1Test"
        value={false}
      />
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "lightblue",
        "borderRadius": 10,
        "elevation": 10,
        "flexDirection": "row",
        "height": 98.57142857142857,
        "justifyContent": "space-between",
        "marginVertical": 8,
        "padding": 10,
        "shadowOffset": {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 5,
        "width": 660,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
          "paddingHorizontal": 10,
        }
      }
      testID="offlineModeTextSettingTest"
    >
      Offline mode
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
      testID="viewSwitch2Test"
    >
      <RCTSwitch
        accessibilityRole="switch"
        disabled={false}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "height": 31,
            "width": 51,
          }
        }
        testID="switch2Test"
        value={false}
      />
    </View>
  </View>
</RCTSafeAreaView>
`;
